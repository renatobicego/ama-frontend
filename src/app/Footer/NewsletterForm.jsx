"use client"
import { Input } from "@/app/utils/materialTailwind"
import { useState } from "react"
import validateEmail from "../utils/formValidation/emailValidation"


const NewsletterForm = ({socialMedia}) => {

    const [email, setEmail] = useState('')

    const handleEmailChange = e => setEmail(e.target.value)

    const handleSubmit = e => {
        e.preventDefault()
        console.log(validateEmail(email))
    }

    return(
        <form onSubmit={handleSubmit} className='flex flex-col justify-between gap-5 h-full w-full lg:w-1/3 parent-btn'>
            <h3 className='text-base md:text-xl font-semibold'>Suscribirse al Newsletter</h3>
            
            <Input 
                size="lg" 
                className="text-white" 
                color="gray" 
                placeholder='Ingrese su Mail*'
                value={email}
                onChange={handleEmailChange}
                />

            <div className='flex gap-8 justify-between items-center'>
                <button className='btn-primary' type="submit">Suscribirse</button>
                <nav>
                    <ul className='flex gap-2 items-center'>
                        {socialMedia.map((item, i) => {
                            return(
                                <a href={item.href} key={i}>
                                    <li><img src={item.logo} alt="" /></li>
                                </a>
                            )
                        })}
                    </ul>
                </nav>
            </div>
        </form>
    )
}

export default NewsletterForm