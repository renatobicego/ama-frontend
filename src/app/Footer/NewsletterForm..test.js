import { cleanup, fireEvent, render, screen } from "@testing-library/react"
import NewsletterForm from "./NewsletterForm"
import '@testing-library/jest-dom/extend-expect'

describe('<NewsletterForm />', () => {

    let form 

    // Mocked socialMedia
    const socialMedia = [
        { href: '', logo: '/test/test.jpg'},
    ]

    // Mock console.log
    const logMock = jest.fn()

    // Replace console.log with logMock
    beforeAll(() => {
        global.console.log = logMock;
    })

    // Restore console.log
    afterAll(() => {
        global.console.log = console._log;
    })

    // Render component
    beforeEach(() => {
        form = render(<NewsletterForm socialMedia={socialMedia}/>)
    })

    afterEach(() => {
        cleanup()
    })

    test('render content', () => {
        expect(form.container).toMatchSnapshot()

        // Is email input field rendered
        const emailInput = screen.getByPlaceholderText('Ingrese su Mail*')
        expect(emailInput).toBeInTheDocument()

        // Is submitButton rendered
        const submitButton = screen.getByText('Suscribirse')
        expect(submitButton).toBeInTheDocument()

    })

    test('form correct value validation', () => {
        expect(form.container).toMatchSnapshot()

        // Find the email input field
        const emailInput = screen.getByPlaceholderText('Ingrese su Mail*')

        // Simulate user input
        fireEvent.change(emailInput, { target: { value: 'mail@mail.com' } })

        // Find the submit button
        const submitButton = screen.getByText('Suscribirse')

        // Simulate form submission
        fireEvent.click(submitButton)

        // Verify that the console.log was called with the expected value
        expect(logMock).toHaveBeenCalledWith(true)
    })

    test('form incorrect value validation', () => {
        expect(form.container).toMatchSnapshot()

        // Find the email input field
        const emailInput = screen.getByPlaceholderText('Ingrese su Mail*')

        // Simulate user input
        fireEvent.change(emailInput, { target: { value: 'invalid_email' } })

        // Find the submit button
        const submitButton = screen.getByText('Suscribirse')

        // Simulate form submission
        fireEvent.click(submitButton)

        // Verify that the console.log was called with the expected value
        expect(logMock).toHaveBeenCalledWith(false)
    })
})