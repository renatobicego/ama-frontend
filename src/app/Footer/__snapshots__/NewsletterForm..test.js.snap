// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsletterForm /> form correct value validation 1`] = `
<div>
  <form
    class="flex flex-col justify-between gap-5 h-full w-full lg:w-1/3 parent-btn"
  >
    <h3
      class="text-base md:text-xl font-semibold"
    >
      Suscribirse al Newsletter
    </h3>
    <div
      class="relative w-full min-w-[200px] h-11"
    >
      <input
        class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-gray-500 text-white"
        placeholder="Ingrese su Mail*"
        value=""
      />
      <label
        class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-blue-gray-400 peer-focus:text-gray-500 before:border-blue-gray-200 peer-focus:before:!border-gray-500 after:border-blue-gray-200 peer-focus:after:!border-gray-500"
      >
         
      </label>
    </div>
    <div
      class="flex gap-8 justify-between items-center"
    >
      <button
        class="btn-primary"
        type="submit"
      >
        Suscribirse
      </button>
      <nav>
        <ul
          class="flex gap-2 items-center"
        >
          <a
            href=""
          >
            <li>
              <img
                alt=""
                src="/test/test.jpg"
              />
            </li>
          </a>
        </ul>
      </nav>
    </div>
  </form>
</div>
`;

exports[`<NewsletterForm /> form incorrect value validation 1`] = `
<div>
  <form
    class="flex flex-col justify-between gap-5 h-full w-full lg:w-1/3 parent-btn"
  >
    <h3
      class="text-base md:text-xl font-semibold"
    >
      Suscribirse al Newsletter
    </h3>
    <div
      class="relative w-full min-w-[200px] h-11"
    >
      <input
        class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-gray-500 text-white"
        placeholder="Ingrese su Mail*"
        value=""
      />
      <label
        class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-blue-gray-400 peer-focus:text-gray-500 before:border-blue-gray-200 peer-focus:before:!border-gray-500 after:border-blue-gray-200 peer-focus:after:!border-gray-500"
      >
         
      </label>
    </div>
    <div
      class="flex gap-8 justify-between items-center"
    >
      <button
        class="btn-primary"
        type="submit"
      >
        Suscribirse
      </button>
      <nav>
        <ul
          class="flex gap-2 items-center"
        >
          <a
            href=""
          >
            <li>
              <img
                alt=""
                src="/test/test.jpg"
              />
            </li>
          </a>
        </ul>
      </nav>
    </div>
  </form>
</div>
`;

exports[`<NewsletterForm /> render content 1`] = `
<div>
  <form
    class="flex flex-col justify-between gap-5 h-full w-full lg:w-1/3 parent-btn"
  >
    <h3
      class="text-base md:text-xl font-semibold"
    >
      Suscribirse al Newsletter
    </h3>
    <div
      class="relative w-full min-w-[200px] h-11"
    >
      <input
        class="peer w-full h-full bg-transparent text-blue-gray-700 font-sans font-normal outline outline-0 focus:outline-0 disabled:bg-blue-gray-50 disabled:border-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-blue-gray-200 border focus:border-2 border-t-transparent focus:border-t-transparent text-sm px-3 py-3 rounded-md border-blue-gray-200 focus:border-gray-500 text-white"
        placeholder="Ingrese su Mail*"
        value=""
      />
      <label
        class="flex w-full h-full select-none pointer-events-none absolute left-0 font-normal peer-placeholder-shown:text-blue-gray-500 leading-tight peer-focus:leading-tight peer-disabled:text-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500 transition-all -top-1.5 peer-placeholder-shown:text-sm text-[11px] peer-focus:text-[11px] before:content[' '] before:block before:box-border before:w-2.5 before:h-1.5 before:mt-[6.5px] before:mr-1 peer-placeholder-shown:before:border-transparent before:rounded-tl-md before:border-t peer-focus:before:border-t-2 before:border-l peer-focus:before:border-l-2 before:pointer-events-none before:transition-all peer-disabled:before:border-transparent after:content[' '] after:block after:flex-grow after:box-border after:w-2.5 after:h-1.5 after:mt-[6.5px] after:ml-1 peer-placeholder-shown:after:border-transparent after:rounded-tr-md after:border-t peer-focus:after:border-t-2 after:border-r peer-focus:after:border-r-2 after:pointer-events-none after:transition-all peer-disabled:after:border-transparent peer-placeholder-shown:leading-[4.1] text-blue-gray-400 peer-focus:text-gray-500 before:border-blue-gray-200 peer-focus:before:!border-gray-500 after:border-blue-gray-200 peer-focus:after:!border-gray-500"
      >
         
      </label>
    </div>
    <div
      class="flex gap-8 justify-between items-center"
    >
      <button
        class="btn-primary"
        type="submit"
      >
        Suscribirse
      </button>
      <nav>
        <ul
          class="flex gap-2 items-center"
        >
          <a
            href=""
          >
            <li>
              <img
                alt=""
                src="/test/test.jpg"
              />
            </li>
          </a>
        </ul>
      </nav>
    </div>
  </form>
</div>
`;
